pkg/storage/ingest/reader_test.go

```diff
@@ -0,0 +1,328 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package ingest
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/dskit/services"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/twmb/franz-go/pkg/kfake"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/pkg/kmsg"
+)
+
+func TestReader(t *testing.T) {
+	const (
+		topicName   = "test"
+		partitionID = 1
+	)
+
+	ctx, cancel := context.WithCancelCause(context.Background())
+	t.Cleanup(func() { cancel(errors.New("test done")) })
+
+	_, clusterAddr := createTestCluster(t, partitionID+1, topicName)
+
+	content := []byte("special content")
+	consumer := newTestConsumer(2)
+
+	startReader(ctx, t, clusterAddr, topicName, partitionID, consumer)
+
+	writeClient := newKafkaProduceClient(t, clusterAddr)
+
+	produceRecord(ctx, t, writeClient, topicName, partitionID, content)
+	produceRecord(ctx, t, writeClient, topicName, partitionID, content)
+
+	messages, err := consumer.waitRecords(2, 5*time.Second, 0)
+	assert.NoError(t, err)
+	assert.Equal(t, [][]byte{content, content}, messages)
+}
+
+func TestReader_IgnoredConsumerErrors(t *testing.T) {
```

Comment:
What's the difference between this test and `TestReader`?

Reply:
oof, that got copied and then not changed. This test is already changed in the other branch I'm working on